connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -740,15 +755,26 @@ private void backoff(long ms) {
         Utils.sleep(ms);
     }
 
+    private void bulkRun(Collection<Callable<Void>> bulk) {
+        try {
+            bulkExecutor.invokeAll(bulk);
+        } catch (InterruptedException e) {
+            // ignore
+        }
+    }
+
     private void startWork() {
         // Start assigned connectors and tasks
         log.info("Starting connectors and tasks using config offset {}", assignment.offset());
+        List<Callable<Void>> callables = new ArrayList<>();
         for (String connectorName : assignment.connectors()) {
-            startConnector(connectorName);
+            callables.add(getConnectorCallable(connectorName, ConnectorStatus.State.RUNNING));
         }
+
         for (ConnectorTaskId taskId : assignment.tasks()) {
-            startTask(taskId);
+            callables.add(getTaskCallable(taskId, TaskStatus.State.RUNNING));
         }
+        bulkRun(callables);
```

Comment:
Hmm, I'm wondering if there's some reason we shouldn't make `addRequest(long, Callable, Callback)` synchronized and remove the synchronization from all the other HTTP call handling methods (`connectors`, `connectorInfo`, `putConnectorConfig`, `requestTaskReconfiguration`, `taskConfigs`, `putTaskConfigs`, `restartConnector`, `restartTask`) which are currently `synchronized` - they pretty much just call out to `addRequest`.